src/transform-sdk/go/transform/processor.go

```diff
@@ -35,12 +36,36 @@ type batchHeader struct {
 	baseSequence         int
 }
 
+type writeEvent struct {
+	record Record
+}
+
+func (e *writeEvent) Record() Record {
+	return e.record
+}
+
+type recordWriter struct {
+	outbuf *rwbuf.RWBuf
+}
+
+func (w *recordWriter) Write(r Record) error {
+	w.outbuf.Reset()
+	r.serializePayload(w.outbuf)
+	b := w.outbuf.ReadAll()
+	// Write the record back out to the broker
+	amt := int(writeRecord(unsafe.Pointer(&b[0]), int32(len(b))))
+	if amt != len(b) {
+		return errors.New("writing record failed with errno: " + strconv.Itoa(amt))
+	}
+	return nil
```

Comment:
It looks like a user transform could receive a record and write multiple back out to the broker. Is that something transform functions could already do, or is it an added capability with this new model?

Reply:
Previously you could return a slice (vector) with multiple records, so no new functionality.

Reply:
cool, that's what I figured 🙂

Reply:
Ah: `[]transform.Record` is a slice. ⛳